com.unity.uiwidgets/Runtime/external/SplayTree.cs

```diff
@@ -120,7 +120,7 @@ class SplayTree<TKey, TValue> : IDictionary<TKey, TValue> where TKey : IComparab
             if (key == null) throw new Exception("should input null");
             if (root == null) throw new Exception("root is null");
             int comp = Splay(key);
-            if (comp < 0) return root.Key;
+            if (comp > 0) return root.Key;
```

Comment:
please ensure this is correct. The implementation of SplayTreeMap in dart is here:
https://github.com/dart-lang/sdk/blob/90ee35cdf9e5faab80cb87690430dcd673b1231b/sdk/lib/collection/splay_tree.dart#L557

Reply:
This guy implement splay tree in opposite way. I don't know why he did this. I can change it by this way or rewrite all his code.

Reply:
got it. let's merge it first and work out a final fix later when we have time